dev-tools/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.9
+FROM python:3.12.0b4-slim
```

Comment:
We're currently on python:3.11-bookworm here, and I think we probably want to stay there.